ansible_collections/arista/cvp/plugins/module_utils/tag_tools.py

```diff
@@ -0,0 +1,238 @@
+                        if state == 'assign':
+                            if auto_create:
+                                self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                                tag_name, tag_val)
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id)
+                        if state == 'unassign':
+                            self.__cv_client.api.tag_assignment_config(element_type,
+                                                                       workspace_id,
+                                                                       tag_name,
+                                                                       tag_val,
+                                                                       device_id,
+                                                                       interface_id,
+                                                                       remove=True)
+                        if mode == 'delete':
+                            self.__cv_client.api.tag_config(element_type, workspace_id,
+                                                            tag_name, tag_val,
+                                                            remove=True)
+
+        # Start build
+        request = 'REQUEST_START_BUILD'
+        reques_id = 'b1'
+        description = 'testing cvprac build'
+        self.__cv_client.api.workspace_config(workspace_id=workspace_id,
+                                              display_name=workspace_name,
+                                              description=description, request=request,
+                                              request_id=reques_id)
+
+        # Check workspace build status and proceed only after it finishes building
+        # XXX: timeout? ansible timeout!
+        b = 0
+        while b == 0:
+            build_id = reques_id
+            try:
+                request = self.__cv_client.api.workspace_build_status(workspace_id,
+                                                                      build_id)
+            except CvpRequestError:
+                continue
+            if request['value']['state'] == 'BUILD_STATE_SUCCESS':
+                b = b + 1
+            else:
+                continue
+
+        api_result = CvApiResult(action_name='tag_' + str(workspace_id))
+        api_result.changed = True
+        api_result.success = True
+
+        # Submit workspace
+        # XXX: timeout=2-3sec!! send status back to user to check status on cvp side
```

Comment:
Strange comment here

Reply:
Same reply as above, we could make this comment more understandable for other people.